docs/content/any/guides/data_filtering_next/index.ruby.md

```diff
@@ -0,0 +1,138 @@
+---
+title: "Filter Data (Preview)"
+weight: 1
+showContentForAnyLanguage: true
+---
+
+# New Data Filtering
+
+When you call `authorize(actor, action, resource)` , Oso evaluates the allow
+rule(s) you have defined in your policy to determine if `actor` is allowed
+to perform `action` on `resource`.  For example, if `jane` wants to `"edit"`
+a `document`, Oso may check that `jane = document.owner`.  But what if you
+need the set of all documents that Jane is allowed to edit?  For example, you
+may want to render them as a list in your application.
+
+One way to answer this question is to take every document in the system and
+call `is_allowed` on it. This isn’t efficient and many times is just
+impossible. There could be thousands of documents in a database but only three
+that have the owner `"steve"`. Instead of fetching every document and passing
+it into Oso, it's better to ask the database for only the documents that
+have the owner `"steve"`. Using Oso to filter the data in your data
+store based on the logic in your policy is what we call “Data Filtering”.
+
+You can use data filtering to enforce authorization on queries made to your data
+store. Oso will take the logic in the policy and turn it into a query for the
+authorized data. Examples could include an ORM filter object, an HTTP request or
+an elastic-search query. The query object and the way the logic maps to a query
+are both user defined.
+
+Data filtering is initiated through two methods on `Oso`.
+
+`{{% exampleGet "authorizedResources" %}}` returns a list of all the
+resources a user is allowed to do an action on. The results of a built and
+executed query.
+
+`{{% exampleGet "authorizedQuery" %}}` returns the query object itself.
+This lets you add additional filters or sorts or any other data to it before
+executing it.
+
+The mapping from polar to a query is defined by an `Adapter`. If an adapter exists for your orm or database you can use it, otherwise you may have to implement your own.
```

Comment:
```suggestion
The mapping from polar to a query is defined by an `Adapter`. If an adapter exists for your ORM or database you can use it, otherwise you may have to implement your own.
```